app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -252,17 +252,17 @@ public void onCreateSessionSafe(Bundle savedInstanceState) {
     public void formSaveCallback(Runnable listener) {
         // note that we have started saving the form
         customFormSaveCallback = listener;
-        interruptAndSaveForm(exit);
+        interruptAndSaveForm(false);
```

Comment:
I am confused on semantics on the new parameter `autoSaving`. Given the `formSaveCallback` is called to auto-save the form as well, should not it be set to true ?

Reply:
Yeah, `autoSaving` as in the `autoSessionPauseSaving`, let me come up with another term.

> Given the formSaveCallback is called to auto-save the form as well, should not it be set to true ?

Considering that the `exit` flag was always set to true for the `formSaveCallback`, I thought about replacing it with another that can convey the same data point. I will review this was well.

Reply:
I changed the flags, added a `sessionExpiredOrSustended` ([3023424](https://github.com/dimagi/commcare-android/pull/2858/commits/30234248cc47e11985f484f4e4d296f2211abc6b)) and `userTriggered` ([312ade0](https://github.com/dimagi/commcare-android/pull/2858/commits/312ade00bdcc22e140d5a8087a15ac7a435958c0))